### PR TITLE
flesh out _guess_arm_aarch64_default_optarch method to lighten up _set_optimal_architecture in gcc.py

### PR DESCRIPTION
**Pull request by boegel**
@geimer I felt the `_set_optimal_architecture` was getting a bit heavy with lots of stuff that only matters for AARCH64, so I fleshed it out which helps with readability imho

### CODE REVIEW

File: easybuild/toolchains/compiler/gcc.py [hunk 1]
```diff
@@ -126,34 +126,42 @@ def _set_optimal_architecture(self, default_optarch=None):
...
-                        # Example: 'mcpu=cortex-a72.cortex-a53' for "ARM Cortex-A53 + Cortex-A72"
-                        default_optarch = 'mcpu=%s' % '.'.join([ct[1] for ct in sorted(core_types, reverse=True)])
-                        self.log.debug("Using architecture-specific compiler optimization flag '%s'", default_optarch)
+                # on AArch64, -mcpu=native is not supported prior to GCC 6,
+                # so try to guess a proper default optarch is none was specified
+                default_optarch = self._guess_arm_aarch64_default_optarch()
 
         super(Gcc, self)._set_optimal_architecture(default_optarch=default_optarch)
+
+    def _guess_arm_aarch64_default_optarch(self):
```
Comment (geimer) [inline easybuild/toolchains/compiler/gcc.py:137]: Nitpicking: Wouldn't `_guess_aarch64_default_optarch` be sufficient?

Comment (boegel) [reply easybuild/toolchains/compiler/gcc.py:137]: sure, I just added the `_arm` bit because we explicitly check for `ARM` before actually using this...

Comment (geimer) [reply easybuild/toolchains/compiler/gcc.py:137]: We actually check for `systemtools.AARCH64` ;)

Comment (boegel) [reply easybuild/toolchains/compiler/gcc.py:137]: @geimer: and for:

```python
if cpu_vendor == systemtools.ARM and cpu_model.startswith('ARM '):
```

Comment (geimer) [reply easybuild/toolchains/compiler/gcc.py:137]: boegel: But that's inside the function, and who knows what other weird heuristics one could come up with...

Comment (boegel) [reply easybuild/toolchains/compiler/gcc.py:137]: OK, sure, I don't feel strongly about it, feel free to rename the method without `_arm`

**Pull request merged**